### ISSUE TITLE
Code should be arranged according to engine module

### ISSUE THREAD

**Issue by MrSapps** [q-gears/q-gears#5]
As per title, it makes it easier to find things and is how the engine should logically fit together.

**Comment by Tom-III** [comment]
I agree, I like the idea of it having the 6 modules and everything being seperate so it's easy to see how things work (and why thigs don't)
